src/engine/channelmixer_autogen.cpp

```diff
@@ -14500,3 +14500,14876 @@ void ChannelMixer::mixChannelsRamping(const EngineMaster::GainCalculator& gainCa
         }
     }
 }
+void ChannelMixer::applyEffectsAndMixChannels(const EngineMaster::GainCalculator& gainCalculator,
+                                              QVarLengthArray<EngineMaster::ChannelInfo*, kPreallocatedChannels>* activeChannels,
+                                              QVarLengthArray<EngineMaster::GainCache, kPreallocatedChannels>* channelGainCache,
+                                              CSAMPLE* pOutput,
+                                              unsigned int iBufferSize,
+                                              unsigned int iSampleRate,
+                                              EngineEffectsManager* pEngineEffectsManager) {
+    int totalActive = activeChannels->size();
+    if (totalActive == 0) {
+        ScopedTimer t("EngineMaster::applyEffectsAndMixChannels_0active");
+        SampleUtil::clear(pOutput, iBufferSize);
+    } else if (totalActive == 1) {
+        ScopedTimer t("EngineMaster::applyEffectsAndMixChannels_1active");
+        CSAMPLE_GAIN newGain[1];
+        EngineMaster::ChannelInfo* pChannel0 = activeChannels->at(0);
+        const int channelIndex0 = pChannel0->m_index;
+        EngineMaster::GainCache& gainCache0 = (*channelGainCache)[channelIndex0];
+        if (gainCache0.m_fadeout) {
+            newGain[0] = 0;
+            gainCache0.m_fadeout = false;
+        } else {
+            newGain[0] = gainCalculator.getGain(pChannel0);
+        }
+        gainCache0.m_gain = newGain[0];
+        CSAMPLE* pBuffer0 = pChannel0->m_pBuffer;
+        SampleUtil::applyGain(pBuffer0, newGain[0], iBufferSize);
+        if (pEngineEffectsManager) {
+            pEngineEffectsManager->process(pChannel0->m_handle, pBuffer0, iBufferSize, iSampleRate, pChannel0->m_features);
+        }
+        for (unsigned int i = 0; i < iBufferSize; ++i) {
+            pOutput[i] = pBuffer0[i];
+        }
+    } else if (totalActive == 2) {
```

Comment:
Honestly I only looked at the for() loop itself - it seems to copy `iBufferSize` items from `pBuffer0` to `pOutput`. A memcpy call will do the same, but in one "big" transaction, so should provide better performance.

Again; I didn't look at the general code around the loop - so I don't know can the whole lot of it be re-worked for performance.